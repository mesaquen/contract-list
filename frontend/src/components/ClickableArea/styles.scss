.clickableArea--container {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1.5rem;
  border: 0.125rem dashed var(--primary-yellow);
  border-radius: var(--roundness);
  cursor: pointer;
  transition: all 0.3s ease-out;

  &:hover {
    background-color: var(--white-hover);
  }
}
