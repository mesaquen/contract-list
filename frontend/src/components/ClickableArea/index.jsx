import React from 'react'
import './styles.scss'

const ClickableArea = ({ children, ...props }) => {
  return (
    <div className="clickableArea--container" {...props}>
      {children}
    </div>
  )
}

export default ClickableArea
