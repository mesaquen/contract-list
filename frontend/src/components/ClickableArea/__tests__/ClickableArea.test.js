import { cleanup, render, fireEvent, screen } from '@testing-library/react'
import ClickableArea from '../index'

beforeEach(cleanup)

const renderComponent = callback => (
  <ClickableArea data-testid="clickablearea" onClick={callback}>
    <span>sample</span>
  </ClickableArea>
)

describe('ClickableArea', () => {
  it('renders correctly', () => {
    const { asFragment } = render(renderComponent())
    expect(asFragment()).toMatchSnapshot()
  })

  it('fires callback when clicked', async () => {
    const callback = jest.fn(() => null)
    render(renderComponent(callback))

    const element = await screen.findByText('sample')
    fireEvent.click(element)

    expect(callback.mock.calls.length).toBe(1)
  })
})
