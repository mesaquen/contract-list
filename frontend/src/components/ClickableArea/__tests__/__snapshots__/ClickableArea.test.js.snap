// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableArea renders correctly 1`] = `
<DocumentFragment>
  <div
    class="clickableArea--container"
    data-testid="clickablearea"
  >
    <span>
      sample
    </span>
  </div>
</DocumentFragment>
`;
